src/mgmt/cli_pool.c

```diff
@@ -353,3 +361,147 @@ dc_pool_destroy(tse_task_t *task)
 	tse_task_complete(task, rc);
 	return rc;
 }
+
+static int
+mgmt_list_pools_cp(tse_task_t *task, void *data)
+{
+	struct mgmt_list_pools_arg	*arg;
+	struct mgmt_list_pools_out	*pc_out;
+	struct mgmt_list_pools_in	*pc_in;
+	uint64_t			 pidx;
+	int				 rc = task->dt_result;
+
+	arg = (struct mgmt_list_pools_arg *)data;
+
+	if (rc) {
+		D_ERROR("RPC error while listing pools: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+	pc_out = crt_reply_get(arg->rpc);
+	D_ASSERT(pc_out != NULL);
+	rc = pc_out->lp_rc;
+	if (rc) {
+		D_ERROR("MGMT_POOL_CREATE replied failed, rc: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+	pc_in = crt_req_get(arg->rpc);
+	D_ASSERT(pc_in != NULL);
+
+	*arg->npools = pc_out->lp_npools;
+
+	/* copy RPC response pools info to client buffer, if provided */
+	if (arg->pools) {
+		/* Response ca_count expected <= client-specified npools */
+		for (pidx = 0; pidx < pc_out->lp_pools.ca_count; pidx++) {
+			struct mgmt_list_pools_one	*rpc_pool =
+					&pc_out->lp_pools.ca_arrays[pidx];
+			daos_mgmt_pool_info_t		*cli_pool =
+					&arg->pools[pidx];
+
+			uuid_copy(cli_pool->mgpi_uuid, rpc_pool->lp_puuid);
+
+			/* allocate rank list for caller (simplifies API) */
+			rc = d_rank_list_dup(&cli_pool->mgpi_svc,
+					     rpc_pool->lp_svc);
+			if (rc) {
+				D_ERROR("Copy RPC reply svc list failed\n");
+				D_GOTO(out_free_svcranks, rc);
+			}
+		}
+	}
+
+	if (arg->pools && (*arg->npools > arg->req_npools)) {
+		D_WARN("pool list contains only client-requested npools=%lu, "
+			"less than npools=%lu in system\n", arg->req_npools,
+			*arg->npools);
+	}
+
+out_free_svcranks:
+	if (arg->pools && (rc != 0)) {
+		for (pidx = 0; pidx < pc_out->lp_pools.ca_count; pidx++) {
+			daos_mgmt_pool_info_t *pool = &arg->pools[pidx];
+
+			if (pool->mgpi_svc)
+				d_rank_list_free(pool->mgpi_svc);
+		}
+	}
+
+out:
+	dc_mgmt_sys_detach(arg->sys);
+	crt_req_decref(arg->rpc);
+
+	return rc;
+}
+
+int
+dc_mgmt_list_pools(tse_task_t *task)
+{
+	daos_mgmt_list_pools_t	        *args;
+	crt_endpoint_t			svr_ep;
+	crt_rpc_t		       *rpc_req = NULL;
+	crt_opcode_t			opc;
+	struct mgmt_list_pools_in      *pc_in;
+	struct mgmt_list_pools_arg	cb_args;
+	int				rc = 0;
+
+	(void) cb_args;
```

Comment:
What does this do?

Reply:
oops - this is left over from a very early form of the function while it was under development. Will remove.